lib/internal/test_runner/test.js

```diff
@@ -648,7 +640,14 @@ class Test extends AsyncResource {
   }
 
   report() {
+    if (this.subtests.length > 0) {
+      this.reporter.plan(this.subtests[0].indent, this.subtests.length);
+    } else {
+      this.reportSubtest();
+    }
+
     let directive;
+    const details = { __proto__: null, duration: this.#duration() };
```

Comment:
Are the changes in this file except the introduction of `details` here necessary? It seems like the other code is being moved around for possibly no gain, but please let me know if that is incorrect.

Reply:
@MoLow might be able to answer this one. I got these changes from his other PR.

Reply:
Same answer, this was to make any yaml changes very simple, we can obviously parse that as well

Reply:
@MoLow my question was about the other changes in this file. Are all of the other changes here related to YAML too?

Reply:
The main addition is adding `reportSubtest` so it can be called multiple times safely (needed here https://github.com/nodejs/node/blob/d97256a32c171e06a9b413e68db5ef12cdec5dbf/lib/internal/test_runner/runner.js#L211 and here https://github.com/nodejs/node/blob/d97256a32c171e06a9b413e68db5ef12cdec5dbf/lib/internal/test_runner/runner.js#L215)

Reply:
Can I resolve this conversation?